File: app/src/main/java/io/givenow/app/helpers/Analytics.java
package io.givenow.app.helpers;

import android.util.Log;

import com.google.android.gms.analytics.HitBuilders;
import com.google.android.gms.analytics.Tracker;
import com.mixpanel.android.mpmetrics.MixpanelAPI;

import org.json.JSONException;
import org.json.JSONObject;

import io.givenow.app.models.ParseUserHelper;

/**
 * Created by aphex on 12/28/15.
 */
public class Analytics {
    public static void sendHit(Tracker tracker, String category, String action, String label) {
        tracker.send(new HitBuilders.EventBuilder().setCategory(category).setAction(action).setLabel(label).build());
    }

    public static void mixpanelTrackIsRegisteredUser(MixpanelAPI mixpanel, String eventName) {
        try {
            JSONObject props = new JSONObject();
            props.put("IsRegistered", ParseUserHelper.isRegistered());
            mixpanel.track(eventName, props);
        } catch (JSONException e) {
            Log.e("MYAPP", "Unable to add properties to JSONObject", e);
        }

    }
}

File: app/src/main/java/io/givenow/app/helpers/Analytics.kt
package io.givenow.app.helpers

import android.util.Log
import com.google.android.gms.analytics.HitBuilders
import com.google.android.gms.analytics.Tracker
import com.mixpanel.android.mpmetrics.MixpanelAPI
import io.givenow.app.models.ParseUserHelper
import org.json.JSONException
import org.json.JSONObject

/**
 * Created by aphex on 12/28/15.

 */
object Analytics {
    fun sendHit(tracker: Tracker, category: String, action: String, label: String) {
        tracker.send(HitBuilders.EventBuilder().setCategory(category).setAction(action).setLabel(label).build())
    }

    fun mixpanelTrackIsRegisteredUser(mixpanel: MixpanelAPI, eventName: String) {
        try {
            val props = JSONObject()
            props.put("IsRegistered", ParseUserHelper.isRegistered)
            mixpanel.track(eventName, props)
        } catch (e: JSONException) {
            Log.e("MYAPP", "Unable to add properties to JSONObject", e)
        }

    }
}
